### PR TITLE
feat(lichess): init

### PR DESCRIPTION
**Pull request by coopw1**
## 🎉 Theme for [lichess.org](https://lichess.org/)🎉
'Lichess is a free and open-source Internet chess server run by a non-profit organization of the same name.' 

mocha
wooo
## 💬 Additional Comments 💬
Forever ago, but was requested [here](https://github.com/catppuccin/catppuccin/discussions/1831) 

## 🗒 Checklist 🗒

- [x] I have read and followed Catppuccin's [submission guidelines](https://github.com/catppuccin/userstyles/blob/main/docs/userstyle-creation.md).
- [x] I have made a new directory underneath `/styles/<name-of-website>` containing the contents of the [`/template`](https://github.com/catppuccin/userstyles/blob/main/template/) directory.
  - [x] I have ensured that the new directory is in **lower-kebab-case**.
  - [x] I have followed the template and kept the preprocessor as [LESS](https://lesscss.org/#overview).
- [x] I have made sure to update the
      [`userstyles.yml`](https://github.com/catppuccin/userstyles/blob/main/src/userstyles.yml)
      file with information about the new userstyle.
- [x] I have included the following files:
  - `catppuccin.user.css` - all the CSS for the userstyle, based on the
    template.
  - `assets/mocha.webp`, `assets/macchiato.webp`, `assets/frappe.webp` and
    `assets/latte.webp` - individual screenshots of the themed website, in all 4
    Catppuccin flavors.
  - `catwalk.webp` - image of all individual screenshots stitched together,
    generated via [catwalk](https://github.com/catppuccin/toolbox#-catwalk).

### CODE REVIEW

**Comment by coopw1** [comment]
sob does lichess have a typo in their html

**Comment by coopw1** [comment]
Thats not a typo smhh

File: styles/lichess.org/catppuccin.user.css [hunk 1]
```diff
@@ -0,0 +1,322 @@
...
+@updateURL https://github.com/catppuccin/userstyles/raw/main/styles/lichess.org/catppuccin.user.css
+@description Soothing pastel theme for lichess.org
+@author Catppuccin
+@license MIT
+
+@preprocessor less
+@var select flavor "Flavor" ["latte:Latte", "frappe:Frappé", "macchiato:Macchiato", "mocha:Mocha*"]
+@var select accentColor "Accent" ["rosewater:Rosewater*", "flamingo:Flamingo", "pink:Pink", "mauve:Mauve", "red:Red", "maroon:Maroon", "peach:Peach", "yellow:Yellow", "green:Green", "teal:Teal", "blue:Blue", "sapphire:Sapphire", "sky:Sky", "lavender:Lavender", "subtext0:Gray"]
+==/UserStyle== */
+@-moz-document url-prefix("https://lichess.org/")
```
Comment (uncenter) [inline styles/lichess.org/catppuccin.user.css:15]:
```suggestion
@-moz-document domain("lichess.org")
```

File: scripts/userstyles.yml [hunk 2]
```diff
@@ -292,6 +292,14 @@ userstyles:
       app-link: "https://github.com/libreddit/libreddit"
       current-maintainers: [*isabelroses]
       past-maintainers: [*andreasgrafen]
+  lichess.org:
```
Comment (uncenter) [inline scripts/userstyles.yml:295]: Hmm. Should we call it just "Lichess"? It seems to be consistently referred to as such on the [about page](https://lichess.org/about):

> In 2010, Thibault Duplessis began work on **Lichess** as a hobby project. The site was simple at the beginning, not even checking to see if moves were legal. He made the site [open source](https://www.youtube.com/watch?v=Tyd0FO0tko8), which means anyone is free to read the source code and make contributions. Gradually, the site improved and collected users as an enthusiastic volunteer staff assembled to help Thibault build and maintain the site.
> 
> Today, **Lichess** users play more than five million games every day. **Lichess** is one of the most popular chess websites in the world while remaining 100% free. Most “free” websites subsist by selling ads or selling user data. Others do it by putting all the good stuff behind paywalls. **Lichess** does none of these things and never will. With no investors demanding profits, **Lichess** staff can focus on improving the site as their only goal.

**Review by uncenter — CHANGES_REQUESTED** [review]

File: scripts/userstyles.yml (hunk 2 above)
Comment (coopw1) [reply scripts/userstyles.yml:295]: Either way is good with me, i was going off the fact that their logo has .org, and how it's called .org in the more header parts of that same article

Comment (coopw1) [reply scripts/userstyles.yml:295]: Actually yeah, just lichess probably makes more sense

**Comment by coopw1** [comment]
I'm heading to sleep, I'll look over stuff again in the morningg

**Comment by coopw1** [comment]
Should I bother changing [.typos.toml)](https://github.com/catppuccin/userstyles/blob/main/.typos.toml) to make it not mark "analyse" as a typo? Or just leave the failed check there

**Comment by uncenter** [comment]
It shouldn't spell-check classes since you can't control those... @nekowinston?

**Comment by nekowinston** [comment]
> It shouldn't spell-check classes since you can't control those... @nekowinston?

Typos isn't syntax-aware AFAIK, so we don't really have control there, I think.

**Comment by nekowinston** [comment]
That fixed it

**Comment by coopw1** [comment]
It looks like I got everything now, think it should be actually ready for review noww

**Comment by isabelroses** [comment]
> It looks like I got everything now, think it should be actually ready for review now

I notice just a few things coming up from the linter, can you fix whatever you can, and anything else we can see later.
Thanks for your work.

**Comment by coopw1** [comment]
> I notice just a few things coming up from the linter, can you fix whatever you can, and anything else we can see later. Thanks for your work.

Yeah, I had looked through those before - they seem to be custom html tags. I'm not really sure how to go about skinning those elements without using them

**Comment by isabelroses** [comment]
> Yeah, I had looked through those before - they seem to be custom html tags. I'm not really sure how to go about skinning those elements without using them

Beyond that, when testing the site I haven't noticed any issues, nor any other expected standards relating to the code.
I also think its best to consult @uncenter and @nekowinston relating the the linter.

**Comment by uncenter** [comment]
> > Yeah, I had looked through those before - they seem to be custom html tags. I'm not really sure how to go about skinning those elements without using them
> 
> Beyond that, when testing the site I haven't noticed any issues, nor any other expected standards relating to the code. I also think its best to consult @uncenter and @nekowinston relating the the linter.

Hmm. I think it's best to disable `selector-type-no-unknown` entirely since it's more hassle to have to ignore every single custom element for every website...

**Comment by nekowinston** [comment]
Now the linter's happy. 🙂
Sorry about this causing delays, but the lint setup is a pretty recent addition to this repo. 😅

File: styles/lichess/catppuccin.user.css [hunk 3]
```diff
@@ -0,0 +1,473 @@
...
+    a:active,
+    a:focus {
+      color: @accent-color;
+    }
+
+    /* Homepage cards */
+    .lobby__app__content,
+    .box {
+      background-color: @mantle;
+    }
```
Comment (isabelroses) [inline styles/lichess/catppuccin.user.css:197]: This should be a fix for the home screen boxes
```suggestion
    }
    .lpools>div {
    	background: fadeout(@crust, 80);
    	border-color: @surface0;
          &:hover {
              background-color: fadeout(@accent-color, 80);
          }
    }
```

**Comment by nekowinston** [comment]
I think the contrast was a bit lacking so I styled the chess pieces & board:
2023-11-21-100842@2x

If you don't agree with some of the color choices, we can use the `#piece(@f, @s)` (`fill`/`stroke`) mixin to change them easily.

File: styles/lichess/catppuccin.user.css [hunk 4]
```diff
@@ -0,0 +1,561 @@
...
+    body {
+      background: @base linear-gradient(to bottom, @crust, @base 116px)
+        no-repeat;
+      color: @text !important;
+    }
+
+    /* lichess button */
+    .site-title a {
+      color: @accent-color;
+    }
```
Comment (isabelroses) [inline styles/lichess/catppuccin.user.css:68]: This should also contain `.site-title span` to contain the .org on the site

**Review by isabelroses — CHANGES_REQUESTED** [review]
I think it would be best if the buttons used crust for text upon hover shown here 
image

File: styles/lichess/catppuccin.user.css [hunk 5]
```diff
@@ -0,0 +1,561 @@
...
+    .orientation-white .files coord:nth-child(2n),
+    .orientation-white .ranks coord:nth-child(2n + 1),
+    .orientation-black .files coord:nth-child(2n + 1),
+    .orientation-black .ranks coord:nth-child(2n) {
+      color: @crust;
+    }
+    square.move-dest {
+      background: radial-gradient(
+        ~"@{accent-color}7F" 19%,
```
[the remark this answers is not in the document]
Comment (coopw1) [reply styles/lichess/catppuccin.user.css:359]: ooh, didn't know what was a thing ty!

File: styles/lichess/catppuccin.user.css (hunk 4 above)
Comment (coopw1) [reply styles/lichess/catppuccin.user.css:68]: I kinda like the way it looks with just lichess being accented, if you guys think both looks better I'm fine with that

**Comment by coopw1** [comment]
> I think it would be best if the buttons used crust for text upon hover shown here image

Done!

**Comment by coopw1** [comment]
> I think the contrast was a bit lacking so I styled the chess pieces & board: 2023-11-21-110719@2x
> 
> If you don't agree with some of the color choices, we can use the `#piece(@f, @s)` (`fill`/`stroke`) mixin to change them easily.

Looks good to me

File: styles/lichess/catppuccin.user.css [hunk 6]
```diff
@@ -0,0 +1,567 @@
+/* ==UserStyle==
+@name lichess Catppuccin
```
Comment (uncenter) [inline styles/lichess/catppuccin.user.css:2]: I think Lichess would be capitalized here?

File: styles/lichess/catppuccin.user.css [hunk 7]
```diff
@@ -0,0 +1,567 @@
...
+@namespace github.com/catppuccin/userstyles/styles/lichess
+@homepageURL https://github.com/catppuccin/userstyles/tree/main/styles/lichess
+@version 0.0.1
+@updateURL https://github.com/catppuccin/userstyles/raw/main/styles/lichess/catppuccin.user.css
+@description Soothing pastel theme for lichess
+@author Catppuccin
+@license MIT
+
+@preprocessor less
+@var select flavor "Flavor" ["latte:Latte", "frappe:Frappé", "macchiato:Macchiato", "mocha:Mocha*"]
```
Comment (uncenter) [inline styles/lichess/catppuccin.user.css:12]: Looks like there is a light/dark/auto theme toggle, `body[data-theme="x"]`. Possible values are "light", "dark", and "system" - you could do something like:

```css
	body[data-theme="light"] {
		:root {
			#catppuccin(@lightFlavor, @accentColor);
		}
	}
	body[data-theme="dark"] {
		:root {
			#catppuccin(@darkFlavor, @accentColor);
		}
	}
	body[data-theme="system"] {
		@media (prefers-color-scheme: light) {
			:root {
				#catppuccin(@lightFlavor, @accentColor);
			}
		}
		@media (prefers-color-scheme: dark) {
			:root {
				#catppuccin(@darkFlavor, @accentColor);
			}
		}
	}

File: styles/lichess/catppuccin.user.css (hunk 4 above)
Comment (isabelroses) [reply styles/lichess/catppuccin.user.css:68]: If your a fan of it perhaps make `.site-title span` text or something else within the CTP theme.

File: styles/lichess/catppuccin.user.css [hunk 8]
```diff
@@ -0,0 +1,567 @@
...
+@var select accentColor "Accent" ["rosewater:Rosewater*", "flamingo:Flamingo", "pink:Pink", "mauve:Mauve", "red:Red", "maroon:Maroon", "peach:Peach", "yellow:Yellow", "green:Green", "teal:Teal", "blue:Blue", "sapphire:Sapphire", "sky:Sky", "lavender:Lavender", "subtext0:Gray"]
+==/UserStyle== */
+@-moz-document domain("lichess.org") {
+  :root {
+    #catppuccin(@flavor, @accentColor);
+  }
```
Comment (coopw1) [inline styles/lichess/catppuccin.user.css:18]: > Looks like there is a light/dark/auto theme toggle, body[data-theme="x"]. Possible values are "light", "dark", and "system" - you could do something like:

```suggestion
@var select lightFlavor "Light Flavor" ["latte:Latte*", "frappe:Frappé", "macchiato:Macchiato", "mocha:Mocha"]
@var select darkFlavor "Dark Flavor" ["latte:Latte", "frappe:Frappé", "macchiato:Macchiato", "mocha:Mocha*"]
@var select accentColor "Accent" ["rosewater:Rosewater*", "flamingo:Flamingo", "pink:Pink", "mauve:Mauve", "red:Red", "maroon:Maroon", "peach:Peach", "yellow:Yellow", "green:Green", "teal:Teal", "blue:Blue", "sapphire:Sapphire", "sky:Sky", "lavender:Lavender", "subtext0:Gray"]
==/UserStyle== */
@-moz-document domain("lichess.org") {
  body[data-theme="light"] {
    :root {
      #catppuccin(@lightFlavor, @accentColor);
    }
  }
  body[data-theme="dark"] {
    :root {
      #catppuccin(@darkFlavor, @accentColor);
    }
  }
  body[data-theme="system"] {
    @media (prefers-color-scheme: light) {
      :root {
        #catppuccin(@lightFlavor, @accentColor);
      }
    }
    @media (prefers-color-scheme: dark) {
      :root {
        #catppuccin(@darkFlavor, @accentColor);
      }
    }
  }
```
I tried doing this, it doesn't seem like it works tho? Not sure what i'm doing wrongg

File: styles/lichess/catppuccin.user.css (hunk 6 above)
Comment (coopw1) [reply styles/lichess/catppuccin.user.css:2]:
```suggestion
@name Lichess Catppuccin
```

File: styles/lichess/catppuccin.user.css (hunk 8 above)
Comment (uncenter) [reply styles/lichess/catppuccin.user.css:18]: Oh yeah lol since the `:root` would be inside the `body[data-theme="dark"]`...

Comment (uncenter) [reply styles/lichess/catppuccin.user.css:18]: Aha! You can do `:root:has(body[data-theme="x"])`!

Comment (coopw1) [reply styles/lichess/catppuccin.user.css:18]: woo!

File: styles/lichess/catppuccin.user.css (hunk 4 above)
Comment (coopw1) [reply styles/lichess/catppuccin.user.css:80]:
```suggestion
    .site-title a {
      color: @accent-color;
    }
    .site-title span {
      color: @text;
    }
```

**Review by isabelroses — APPROVED** [review]

**Review by uncenter — APPROVED** [review]

File: styles/lichess/catppuccin.user.css [hunk 9]
```diff
@@ -0,0 +1,582 @@
+/* ==UserStyle==
+@name Lichess Catppuccin
+@namespace github.com/catppuccin/userstyles/styles/lichess
+@homepageURL https://github.com/catppuccin/userstyles/tree/main/styles/lichess
+@version 0.0.1
+@updateURL https://github.com/catppuccin/userstyles/raw/main/styles/lichess/catppuccin.user.css
+@description Soothing pastel theme for lichess
```
Comment (uncenter) [inline styles/lichess/catppuccin.user.css:7]: Oop I guess make Lichess capitalized here too
```suggestion
@description Soothing pastel theme for Lichess
```

**Comment by uncenter** [comment]
LGTM!

**Pull request merged**